src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -274,6 +275,40 @@ internal BinaryClassificationTrainers(BinaryClassificationCatalog catalog)
                 Evaluate(x.Scores, labelColumn), x.Scores, x.Fold)).ToArray();
         }
 
+        /// <summary>
+        /// Change threshold for binary model.
+        /// </summary>
+        /// <typeparam name="TModel">An implementation of the <see cref="IPredictorProducing{TResult}"/></typeparam>
+        /// <param name="chain">Chain of transformers.</param>
+        /// <param name="threshold">New threshold.</param>
+        /// <returns></returns>
+        public TransformerChain<BinaryPredictionTransformer<TModel>> ChangeModelThreshold<TModel>(TransformerChain<BinaryPredictionTransformer<TModel>> chain, float threshold)
```

Comment:
I am not sure if this should be a new function. Could we add a parameter, `threshold`, to all binary trainers?

Reply:
Ok, we can add that as parameter to binary trainer. Question is if you train your model, how you gonna change threshold? Retrain model?
I think this method has right to live.

---
In reply to: [266062138](https://github.com/dotnet/machinelearning/pull/2969#discussion_r266062138) [](ancestors = 266062138)

Reply:
Retrain looks fine to me. I really don't feel adding a helper function is a good idea. This is not a Transformer, so I expect it will become a orphan in the future. Like FFM, PFI and so on don't care about it because it's not a standard binary classifier.

---
In reply to: [266088129](https://github.com/dotnet/machinelearning/pull/2969#discussion_r266088129) [](ancestors = 266088129,266062138)

Reply:
> I am not sure if this should be a new function. Could we add a parameter, `threshold`, to all binary trainers? #Pending

Historically we have found that adding options to "all" trainers just invites inconsistency and is a nightmare from a maintainability perspective. For those reasons we no longer do that. So I strongly object to that.

Reply:
As I think about it more, there's something about this idea of getting `ITransformer` implementors from existing `ITransformer` implementors that I find very appealing. Not just for this (which is a worthy use of this idea), but many other scenarios as well.

So for example, certain regressor algorithms are parametric w.r.t. their labels (in fact, most are). But there's a problem with merely normalizing the label, because then the predicted label is according to that same scale. In `sklearn` you could accomplish this fairly easily via the [`inverse_transform` method on their equivalent of what we call a normalizer, the `StandardScalar`](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html#sklearn.preprocessing.StandardScaler). So imagine you could get from a `NormalizerTransformer` another `NormalizerTransformer` that provides the inverse offset and scaling for any affine normalization, and whatnot. That would be pretty nice, would it not be?

So far from discouraging this pattern, I think we should do more of it.